### PR TITLE
Allow label sets initialization while registering metrics

### PR DESCRIPTION
**Pull request by davidbrusius**
`init_label_sets` is a new config that expects a Hash with metrics names as keys and an array of label sets as values.

Example:

```ruby
{
  metric_name: [{label: 'value1', other_label: 'value1'}, {label: 'value1', other_label: 'value2'}],
  another_metric_name: [{label: 'value1', other_label: 'value1'}]
}
```

Label sets are fetched during metric registration and initialized by calling [metric.init_label_set](https://github.com/prometheus/client_ruby#init_label_set).

This is specially useful for initializing global metrics since they are registered upon leader election.

### CODE REVIEW

File: lib/sidekiq_prometheus/job_metrics.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def call(worker, job, queue)
 
       result
     rescue => e
-      if e.class.to_s == "Sidekiq::Limiter::OverLimit"
+      if e.instance_of?(::Sidekiq::Limiter::OverLimit)
```
Comment (davidbrusius) [inline lib/sidekiq_prometheus/job_metrics.rb:33]: This was autofixed by `rake standard:fix`.

**Comment by davidbrusius** [comment]
👋🏻 Hi @leklund, thoughts on this one? We are using the `sidekiq_queue_latency` metric for autoscaling purposes and it would be really helpful to have label initialization for this global metric.

Thanks!

File: README.md [hunk 2]
```diff
@@ -79,6 +79,7 @@ end
 * `custom_labels`: Hash of metrics and labels that can be applied to specific metrics. The metric name must be a registered metric. `Hash{Symbol (metric name) => Array<Symbol> (label names)}`
 * `gc_metrics_enabled`: Boolean that determines whether to record object allocation metrics per job. The default is `true`. Setting this to `false` if you don't need this metric.
 * `global_metrics_enabled`: Boolean that determines whether to report global metrics from the PeriodicMetrics reporter. When `true` this will report on a number of stats from the Sidekiq API for the cluster. This requires Sidekiq::Enterprise as the reporter uses the leader election functionality to ensure that only one worker per cluster is reporting metrics.
+* `init_label_sets`: Hash of metrics and label sets that are initialized by calling `metric.init_label_set` after the metric is registered. `Hash{Symbol (metric name) => Array<Hash> (label sets)}`.
```
Comment (leklund) [inline README.md:82]:
```suggestion
* `init_label_sets`: Hash of metrics and label sets that are initialized by calling `[metric.init_label_set](https://github.com/prometheus/client_ruby/tree/e144d6225d3c346e9a4dd0a11f41f8acde386dd8#init_label_set)` after the metric is registered. `Hash{Symbol (metric name) => Array<Hash> (label sets)}`.
```

**Comment by leklund** [comment]
@davidbrusius I had one comment to add a link to the prometheus client docs to make it easier to understand what init_label_sets will do. Looks great otherwise! Thanks!

**Comment by leklund** [comment]
Oops, my suggested change wasn't valid markdown. Fixing

File: README.md [hunk 3]
```diff
@@ -79,6 +79,7 @@ end
 * `custom_labels`: Hash of metrics and labels that can be applied to specific metrics. The metric name must be a registered metric. `Hash{Symbol (metric name) => Array<Symbol> (label names)}`
 * `gc_metrics_enabled`: Boolean that determines whether to record object allocation metrics per job. The default is `true`. Setting this to `false` if you don't need this metric.
 * `global_metrics_enabled`: Boolean that determines whether to report global metrics from the PeriodicMetrics reporter. When `true` this will report on a number of stats from the Sidekiq API for the cluster. This requires Sidekiq::Enterprise as the reporter uses the leader election functionality to ensure that only one worker per cluster is reporting metrics.
+* `init_label_sets`: Hash of metrics and label sets that are initialized by calling `[metric.init_label_set](https://github.com/prometheus/client_ruby/tree/e144d6225d3c346e9a4dd0a11f41f8acde386dd8#init_label_set)` after the metric is registered. `Hash{Symbol (metric name) => Array<Hash> (label sets)}`.
```
Comment (leklund) [inline README.md:82]:
```suggestion
* `init_label_sets`: Hash of metrics and label sets that are initialized by calling `metric.init_label_set` after the metric is registered. See[ prometheus-client docs](https://github.com/prometheus/client_ruby/tree/e144d6225d3c346e9a4dd0a11f41f8acde386dd8#init_label_set) for details. `Hash{Symbol (metric name) => Array<Hash> (label sets)}`.
```

**Review by leklund — APPROVED** [review]

**Comment by davidbrusius** [comment]
Appreciate the help on this one @leklund

**Pull request merged**